Update DVDInputStreamBluray for UHD BD

fandangos2 opened the pull request:
## Description
Fix: WARNING: CDVDInputStreamBluray - unhandled libbluray event 32
Fix: WARNING: CDVDInputStreamBluray - unhandled libbluray event 33

Set Bluray Player profile to accomodate UHD BDs.
Set Bluray Player UHD capabilities
Set HDR Preference. If HDR metadata is present it should be used and if the display doesn't support it, it will tonemap to SDR.

## Motivation and Context
Now that libbluray was updated to 1.1.2 two new events are added
BD_EVENT_KEY_INTEREST_TABLE and BD_EVENT_UO_MASK_CHANGED, those would appear as unhandled in Kodi's logs.

Updated the player profile that is requested when playing a disc to the newest one: profile 6v3.1.
If disc requests if the player supports 4K, libbluray will report as yes, even if the screen is 1080p Kodi will downscale normaly.
If disc requests if the player support HDR, libbluray will report as yes and even if the screen does not, Kodi will tonemap normaly. 

## How Has This Been Tested?
This has been tested on a Windows x64 build.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

fandangos2 committed:
Update DVDInputStreamBluray for UHD BD

Fix:
WARNING: CDVDInputStreamBluray - unhandled libbluray event 32
Fix:
WARNING: CDVDInputStreamBluray - unhandled libbluray event 33

Set Bluray Player profile to accomodate UHD BDs.
Set Bluray Player UHD capabilities
Set HDR Preference. If HDR metadata is present it should be used and if the display doesn't support it, it will tonemap to SDR.

arnova commented:
@Paxxi : Sorry, not really my turf so unsure whether this is correct.

Paxxi commented:
Ok, thanks anyway 😀

jjd-uk commented:
Maybe @ace20022 can take a look, although he hasn't been active in a long while.

cpmw167 commented:
I'm not a Team Kodi member, but this PR is definitely correct. It just bumps the player profile from v5 (3D support) to v6 (UHD support), adds two missing events and set players capabilities/registers for UHD support.

Although *theoretically* we shouldn't set all capabilities as enabled, because some of them are reserved and also because it enables all capabilities like e.g. HDR10+ and Dolby Vision which are not supported in Kodi. However, I don't believe it makes any difference in libbluray though.

fandangos commented:
Hi Cpmw, thanks for the pull request for encrypted discs, very nice finding.

Dolby Vision is supported by Kodi on Nvidia Shield Pro and the Apple TV but only the first layer, not the enhancement layer.
I played with those settings and it just enables those on libbluray and actually DVDDemuxFFmpeg handles most of the stream selection.
Example I could not stop the dolby vision enhancement layer from being displayed so I dropped it with dvddemuxffmpeg on another PR.

This is all from my debugging. I bet someone with more experience can fill in the remaining info.

Paxxi commented:
Thank you @cpmw167 it's great to get feedback on this.

I believe we should be ok getting this in for v19 unless ace have some feedback.

I was concerned that we bump the minimum required version of libbluray but as all platforms build I think we're good(@fandangos I didn't want to say anything until I was sure 😀)

Rechi reviewed (commented):
Currently libbluray >= 0.9.3 is required, with this PR libbluray >= 1.0.2 will be required.
If the required version is increased it has to be adjusted in the CMake find module and the ifdefs for older libbluray versions have to be removed too.

fandangos2 commented:
@Rechi I have updated all the strings that returned from 'BLURAY_VERSION_CODE' and I have removed the clog output that it can't play profile 6 bd-j because now it can. 

Since libbluray comes pre-compiled, I don't think I can update the CMAKE find module.

Rechi commented on xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamBluray.cpp:
```diff
@@ -549,6 +549,10 @@ void CDVDInputStreamBluray::ProcessEvent() {
   case BD_EVENT_SECONDARY_VIDEO_SIZE:
   case BD_EVENT_SECONDARY_VIDEO_STREAM:
   case BD_EVENT_PLAYMARK:
+#if (BLURAY_VERSION > BLURAY_VERSION_CODE(1,0,2))
```
Why? Those are available since 0.8.1.

Rechi commented on xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamBluray.cpp:
```diff
@@ -1131,10 +1135,15 @@ void CDVDInputStreamBluray::SetupPlayerSettings()
-  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_3D_CAP, 0xffffffff);
+#if (BLURAY_VERSION > BLURAY_VERSION_CODE(1,0,2))  
+  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_PLAYER_PROFILE, BLURAY_PLAYER_PROFILE_6_v3_1);
+  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_3D_CAP, 0xffffffff);	  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_3D_CAP, 0xffffffff);
```
Why move this into a `BLURAY_VERSION > BLURAY_VERSION_CODE(1, 0, 2)` ifdef?

Rechi commented on xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamBluray.cpp:
```diff
@@ -1131,10 +1135,15 @@ void CDVDInputStreamBluray::SetupPlayerSettings()
+  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_UHD_DISPLAY_CAP, 0xffffffff);
+  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_HDR_PREFERENCE, 0xffffffff);
+#else
+bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_PLAYER_PROFILE, BLURAY_PLAYER_PROFILE_5_v2_4);
```
> Use spaces as tab policy with an indentation size of 2.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation

Rechi commented on xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamBluray.cpp:
```diff
@@ -1131,10 +1135,15 @@ void CDVDInputStreamBluray::SetupPlayerSettings()
     CLog::Log(LOGWARNING, "CDVDInputStreamBluray::Open - Blu-ray region must be set in setting, assuming region A");
     region = BLURAY_REGION_A;
   }
-  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_REGION_CODE, static_cast<uint32_t>(region));
```
Is this not needed anymore?

fandangos2 commented:
@rechi, sorry made a small confusion here. It should be corrected now.

Rechi commented on xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamBluray.cpp:
```diff
@@ -1133,8 +1135,15 @@ void CDVDInputStreamBluray::SetupPlayerSettings()
   bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_PARENTAL, 99);
-  bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_PLAYER_PROFILE, BLURAY_PLAYER_PROFILE_5_v2_4);
   bd_set_player_setting(m_bd, BLURAY_PLAYER_SETTING_3D_CAP, 0xffffffff);
+#if (BLURAY_VERSION >= BLURAY_VERSION_CODE(1,0,2))  
```
> Commas have to be followed by one space.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#34-whitespace

fandangos2 replied:
Done

Rechi reviewed (commented):
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

Paxxi commented:
jenkins build this please

Paxxi commented:
Jenkins is being annoying but this should be good to go.
